tasks/internal/check-signing-certificates/check-signing-certificates.yaml

```
@@ -0,0 +1,74 @@
+---
+apiVersion: tekton.dev/v1
+kind: Task
+metadata:
+  name: check-signing-certificates
+  annotations:
+    tekton.dev/pipelines.minVersion: "0.12.1"
+    tekton.dev/tags: release
+spec:
+  description: |-
+    Tekton task to verify the expiration of signing certificates (UMB and Pyxis)
+  params:
+    - name: umb_ssl_cert_secret_name
+      type: string
+      description: Kubernetes secret name for UMB
+    - name: umb_ssl_cert_file_name
+      type: string
+      description: The name of the file with the UMB certificate in the secret
+    - name: pyxis_ssl_cert_secret_name
+      type: string
+      description: Kubernetes secret name for Pyxis
+    - name: pyxis_ssl_cert_file_name
+      type: string
+      description: The name of the file with the Pyxis certificate in the secret
+  volumes:
+    - name: umb-ssl-cert-secret
+      secret:
+        secretName: $(params.umb_ssl_cert_secret_name)
+        optional: false
+    - name: pyxis-ssl-cert-secret
+      secret:
+        secretName: $(params.pyxis_ssl_cert_secret_name)
+        optional: false
+  steps:
+    - name: check-certificates
+      image: quay.io/konflux-ci/release-service-utils:7072b5b5aa2cc0fc44baf8fb9f8e4a3540fc4fed
+      computeResources:
+        limits:
+          memory: 128Mi
+        requests:
+          memory: 128Mi
+          cpu: 100m
+      volumeMounts:
+        - name: umb-ssl-cert-secret
+          mountPath: /mnt/umb_ssl_cert_secret
+          readOnly: true
+        - name: pyxis-ssl-cert-secret
+          mountPath: /mnt/pyxis_ssl_cert_secret
+          readOnly: true
+      script: |
+        #!/usr/bin/env bash
+        set -ex
+
+        echo "=== Checking signing certificate expiration ==="
+
+        # Check UMB certificate
+        UMB_CERT_FILE="/mnt/umb_ssl_cert_secret/$(params.umb_ssl_cert_file_name)"
+        echo "Checking UMB certificate: ${UMB_CERT_FILE}"
+        if ! check_cert_expiration "${UMB_CERT_FILE}"; then
```

You have several exit code scenarios. 

https://github.com/konflux-ci/release-service-utils/blob/main/utils/check_cert_expiration#L9

If the cert is expiring soon, you'd fail the pipeline here instead of warn.